Components/Table/Table.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import {TableHeader} from './TableHeader';
+import {TableBody} from './TableBody';
+import {TableFooter} from './TableFooter';
+import styled from 'styled-components';
+
+export interface Column {
+    title?: string;
+    placeholder?: string;
+    alignText?: string;
+    value?: any;
+    sortProperty?: string;
+}
+
+export interface TableProps {
+    columns: Column[];
+    data: any[];
+    headerColumnClicked?: (val: Column) => void;
+    style?: object;
+    className?: string;
+}
+
+const TableWrapper = styled.table`
+    width: 100%;
+    border-collapse: collapse;
+    border: 1px solid lightgray
+`
+
+export class Table extends React.Component<TableProps> {
```

Comment:
@TravisSalad Just so you know, you can use stateless components in Typescript too if you want.

```tsx
export function Table(props: TableProps) {}
```